### PR TITLE
Change extends 'base.html' to extends 'generic/object_detail.html' for detail templates

### PR DESCRIPTION
**Pull request by briddo**
# About

Fixes: #129 

## Additional

- [x] Use "Software" as plural in templates breadcrumbs

### CODE REVIEW

**Comment by jvanderaa** [comment]
@briddo what is the status on this PR?

**Comment by jvanderaa** [comment]
Attempting to restart tests.

**Review by glennmatthews — COMMENTED** [review]
Probably this PR doesn't go far enough - for example the `block header` in most of these templates may now be 100% redundant with the default block provided by `object_detail.html` and can/should be removed if that is the case.

**Comment by gsnider2195** [comment]
None of these files are being used anymore. Can we go ahead and delete them?

**Comment by gsnider2195** [comment]
> None of these files are being used anymore. Can we go ahead and delete them?

Sorry, `nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/softwarelcm.html` is still being used. The rest of the html files as well as `nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/hardwarelcm_delete.html` appear to be no longer used and can be safely deleted?

**Comment by gsnider2195** [comment]
I updated these. One thing I noticed is the default `verbose_name_plural` is causing some weirdness on a few pages for `Software` and `Validated Software`:

image

Do we want to change those as part of this PR? This will introduce a new migration.

**Comment by gsnider2195** [comment]
> @gsnider2195 Would changing `verbose_name_plural` affect GraphQL schema?

I believe it will. You can see on the demo instance that GraphQL uses `softwares` and `validated_softwares`

**Comment by progala** [comment]
> > @gsnider2195 Would changing `verbose_name_plural` affect GraphQL schema?
> 
> I believe it will. You can see on the demo instance that GraphQL uses `softwares` and `validated_softwares`

Thanks. What values do you propose `verbose_name_plural` to be for `Software` and `Validated Software` ?

**Comment by gsnider2195** [comment]
> > > @gsnider2195 Would changing `verbose_name_plural` affect GraphQL schema?
> > 
> > 
> > I believe it will. You can see on the demo instance that GraphQL uses `softwares` and `validated_softwares`
> 
> Thanks. What values do you propose `verbose_name_plural` to be for `Software` and `Validated Software` ?

`software` and `validated_software`

**Comment by progala** [comment]
> > > > @gsnider2195 Would changing `verbose_name_plural` affect GraphQL schema?
> > > 
> > > 
> > > I believe it will. You can see on the demo instance that GraphQL uses `softwares` and `validated_softwares`
> > 
> > 
> > Thanks. What values do you propose `verbose_name_plural` to be for `Software` and `Validated Software` ?
> 
> `software` and `validated_software`

I tested it out locally, by changing `verbose_name_plural` for `Software` to `software`. This seems to result in the GraphQL schema where `Query.software [SoftwareLCMType]` shadows `Query.software SoftwareLCMType`, i.e. `Query.software SoftwareLCMType` is no longer accessible.

**Comment by bryanculver** [comment]
> > > > > @gsnider2195 Would changing `verbose_name_plural` affect GraphQL schema?
> > > > 
> > > > 
> > > > I believe it will. You can see on the demo instance that GraphQL uses `softwares` and `validated_softwares`
> > > 
> > > 
> > > Thanks. What values do you propose `verbose_name_plural` to be for `Software` and `Validated Software` ?
> > 
> > 
> > `software` and `validated_software`
> 
> I tested it out locally, by changing `verbose_name_plural` for `Software` to `software`. This seems to result in the GraphQL schema where `Query.software [SoftwareLCMType]` shadows `Query.software SoftwareLCMType`, i.e. `Query.software SoftwareLCMType` is no longer accessible.

@snaselj is going to see if in the short-term we can override the template to make it say "Software" instead of "Softwares" and I'll open an issue to make the change to the `verbose_name_plural` sometime in the future.

**Comment by snaselj** [comment]
Rebased on top of `develop`

**Comment by snaselj** [comment]
Implemented better way to show "Software" even with search forms.

File: nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/softwarelcm.html [hunk 1]
```diff
@@ -1,130 +1,73 @@
...
-        </div>
-    </div>
-    <div class="pull-right noprint">
-        {% plugin_buttons object %}
-        {% if perms.nautobot_device_lifecycle_mgmt.add_softwareimagelcm and active_tab == 'software-images' %}
-            <a href="{% url 'plugins:nautobot_device_lifecycle_mgmt:softwareimagelcm_add' %}?software={{ object.pk }}" class="btn btn-success">
-                <span class="mdi mdi-plus-thick" aria-hidden="true"></span>
-                Add a Software Image
-            </a>
-        {% endif %}
```
Comment (progala) [inline nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/softwarelcm.html:37]: Can we replicate functionality provided by this code? 

Proposed changes remove the `Add a Software Image` button.

Before:

image


After:

image

**Review by progala — CHANGES_REQUESTED** [review]
Looks great, thanks for the updates! 

There's just one thing that needs fixing, by using the new template we are losing `Add a Software Image` button on the `Software Images` tab in the Software view. Could we have it reinstated?

File: nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/softwarelcm.html (hunk 1 above)
Comment (snaselj) [reply nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/softwarelcm.html:37]: Fixed, thanks for catching this @progala

Comment (progala) [reply nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/softwarelcm.html:37]: Thanks @snaselj , looking great!

**Review by progala — APPROVED** [review]

**Pull request merged**